### DISCUSSION TITLE
Parts ingredients

### DISCUSSION THREAD

**Discussion by kaare** [ledgersmb/LedgerSMB#6964] [Design]
We need to store the ingredient list for most of our items. There are some issues here
1. We keep the original ingredients list (often in English) but translate to Danish. So this looks like a new table with parts id and language id as foreign keys and a text field
2. The list is used for printing labels. So we would probably like to have some formatting information embedded.

**Comment by ylavoie** [comment]
So we need to open the possibility to have Transifex portions for user data. Interesting.

**Comment by einhverfr** [comment]
I am not sure transferex is the right solution here.  More likely the same manual translation processes we use for part names, project names, and other user supplied data would be used.

**Comment by ehuelsmann** [comment]
@kaare , sorry to be getting back on this topic after so long. I think I have a solution: I'm thinking of creating the ability to add an arbitrary number of custom text fields to parts. These can be named by the user and filled with their own content. These would be available for printing by making them available as a print template variable. Would that solve your use-case?

**Comment by kaare** [comment]
Hi

My personal use case is for labels that probably would need bold and italics, font size information, and also page formatting - i.e. how many labels that can fit across and as a whole on a page.

That's very specific, and I would be happy to make the necessary code to print my labels, but I think I would need some metadata to do so, as well as a hook to fit the print process.

**Comment by hasorli** [comment]
I  have a user case with label printing of parts labels.  Custom text fields to parts can probably solve this.

**Comment by ehuelsmann** [comment]
@kaare @hasorli you could help me achieve this functionality if you could create a wireframe or mock-up of the current Part page with this functionality drawn in. That makes it easier for me to start working on this out of the blue. I've got it planned for 1.10, but there's lots of other stuff that I want in there too, so a bit of UX help would be more than welcome!
